azavea/raster-foundry
# Wire up import modal

##### Installation
Run `npm.sh install`, `bundle.sh --vendor` and `manage.sh migrate`

##### Testing
 * Create a new import. Upon success, the modal should be hidden. Check that the database is updated correctly for all fields. Try out different options of dropdowns. 
 * After clicking the import button after a successful submission, all fields should be clear.
 * After closing the import modal before a successful submission, and then opening it, the fields should retain their values.
 * Try to hit all client-side and server-side error conditions using the validation functions as a guide. When using an existing layer name for a user, you should be brought back to Pane-2 and the error message will persist until changing the layer name.
 * Make sure that back buttons work.

##### Known Issues
 * The following inputs need further guidance from Management/Design, and don't currently work: Pane-2: Source Projection and Pane-3: Source SRS
 * The style of the box around the tags needs some help from Design.
 * The only server-side error that we can deal with is duplicate_name. I haven't been able to find any other server-side errors that are possible given our level of client-side validation, but it's probably possible. Maybe we should have a catch-all error field for any other server-side messages on Pane-3?
 * The progress of uploading is only shown in the console. Showing it in the GUI is left to #104 
  
Connects #47
